sphinx/ext/todo.py

```diff
@@ -110,9 +110,12 @@ def process_todo_nodes(app, doctree, fromdocname):
 
         for todo_info in env.todo_all_todos:
             para = nodes.paragraph(classes=['todo-source'])
-            description = _('(The <<original entry>> is located in '
-                            ' %s, line %d.)') % \
-                          (todo_info['source'], todo_info['lineno'])
+            if app.config['todo_link_only']:
+                description = _('<<original entry>>')
```

Comment:
I think that the best would be either the link alone or the link with something like `package.module` or `package.module:{class,function,method}` depending where the `todo` is. This would make sense for the user (who doesn't care where the code is, but if there is some functionality he should take with care) and for the developer (who likely knows where a module is implemented).
The relative path from `conf.py` doesn't say much, as it depends where the file is. Plus in principle you can have it the documentation in a repository different from the code.